src/preset.ts

```diff
@@ -324,6 +327,45 @@ export interface TestPreset extends Preset {
     __generator?: string; // Getting this from the config preset
 }
 
+export interface PackageOutputOptions {
+    debug?: boolean;
+    verbose?: boolean;
+}
+
+export interface PackagePreset extends Preset {
+    configurePreset?: string;
+    inheritConfigureEnvironment?: boolean; // Defaults to true
+    configurations?: string[];
+    generators?: string[];
+    variables?: { [key: string]: string | null | undefined };
+    configFile?: string;
+    output?: PackageOutputOptions;
+    packageName?: string;
+    packageVersion?: string;
+    packageDirectory?: string;
+    vendorName?: string;
+
+    // Private fields
+    __binaryDir?: string; // Getting this from the config preset
+    __generator?: string; // Getting this from the config preset
+}
+
+export interface WorkflowStepsOptions {
+    type: string;
+    name: string;
+}
+
+export interface WorkflowPreset extends Preset {
+    name: string;
+    displayName?: string;
+    description?: string;
+    vendor?: VendorType;
+    steps: WorkflowStepsOptions[];
+    // Private fields
```

Comment:
What are these for? These aren't part of the docs for WorkflowPreset, so I'm wondering if there is another way that we can do this?

Reply:
I later added functionality to look at every workflow step __binaryDir and __generator and didn't need these variables defined globally for a workflow (although, all the workflow steps need to have the same associated configure preset, meaning same __binaryDir and __generator for all, so we could still have one set of these variables defined globally). I removed them now.